Services/Certificate/classes/LearningProgress/class.ilCertificateCourseLearningProgressEvaluation.php

```diff
@@ -1,48 +1,23 @@
-<?php
+<?php declare(strict_types=1);
 /* Copyright (c) 1998-2018 ILIAS open source, Extended GPL, see docs/LICENSE */
 
 /**
  * @author  Niels Theen <ntheen@databay.de>
  */
 class ilCertificateCourseLearningProgressEvaluation
 {
-    /**
-     * @var ilCertificateTemplateRepository
-     */
-    private $templateRepository;
-
-    /**
-     * @var ilSetting
-     */
-    private $setting;
-
-    /**
-     * @var ilCertificateObjectHelper
-     */
-    private $objectHelper;
+    private ilCertificateTemplateRepository $templateRepository;
+    private ?ilSetting $setting;
+    private ?ilCertificateObjectHelper $objectHelper;
+    private ?ilCertificateLPStatusHelper $statusHelper;
+    private ?ilCertificateObjUserTrackingHelper $trackingHelper;
```

Comment:
These types are not optional. Only the passed constructor arguments are optional.